src/data_source/mod.rs

```diff
@@ -170,6 +177,43 @@ impl<'env> Connection<'env> {
     }
 }
 
+impl<'env> Connection<'env, AutocommitOff> {
+    /// Set autocommit mode off, triggers implicit rollback of any running transaction
+    pub fn enable_autocommit(mut self) -> Return<Connection<'env, AutocommitOn>, Self> {
+        match self.handle.rollback() {
```

Comment:
Is this somehow required by the ODBC API or would this be an additional invariant enforced by odbc-safe (which it tries to avoid AFAIU)?

Reply:
Actually I've inserted this because I think this is more expected than default ODBC behavior. According to [docs](https://docs.microsoft.com/ru-ru/sql/relational-databases/native-client/odbc/performing-transactions-in-odbc?view=sql-server-2017) when we switch to autocommit mode in the middle of transaction ODBC default behavior is to implicitely _commit_ any running transactions which can potentially lead to data corruption. Automatic rollback looks much safer default.

Reply:
While I agree that it is the more sensible behaviour, I think odbc-safe tries to avoid changing ODBC semantics as much as possible, i.e.

> Not to abstract away any power of the underlying API

Hence I think the ODBC behaviour should be exposed and wrappers around this library can add saner semantics if they want to.

(Thinking about this some more, this probably also applies to the rollback before disconnect since the current solution takes away the power to see that error upon disconnect when one forgot to commit/rollback a transaction.)

Reply:
You're right this logic should probably belong to odbc-rs crate, not odbc-safe, I'll fix this